frontend/public/components/utils/k8s-watch-hook.ts

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector, useDispatch } from 'react-redux';
+import { Map as ImmutableMap } from 'immutable';
+import { createSelectorCreator, defaultMemoize } from 'reselect';
+import { makeQuery, makeReduxID } from './k8s-watcher';
+import * as k8sActions from '../../actions/k8s';
+import { K8sResourceKind, K8sKind } from '../../module/k8s';
+import { FirehoseResource } from './types';
+import { RootState } from '../../redux';
+
+export class NoModelError extends Error {
+  constructor() {
+    super('Model does not exist');
+  }
+}
+
+export const useK8sWatch = <R extends K8sResourceKind = K8sResourceKind>(
+  resources: FirehoseResource[] = [],
+): WatchK8sResult<R> => {
+  const k8sModelSelectorCreator = React.useMemo(
+    () =>
+      createSelectorCreator(
+        //specifying createSelectorCreator<ImmutableMap<string, any>> throws type error
```

Comment:
The `as ..` typecast is basically a band-aid to satisfy the `createSelectorCreator` signature, right?

I can look into this, ideally we can avoid the typecast altogether, but how is this better than a simple `as any` ?

Reply:
yes, that is correct. I think there's no advantage over simple `as any`. Im using `as any` now since its shorter :)

Reply:
I've looked into this in my other PR, check out https://github.com/openshift/console/pull/4211#discussion_r388453175.

My conclusion is

> It seems that Reselect typings for `createSelectorCreator` API aren't robust enough.

so using `as any` currently seems to be the simplest work-around.